service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java

```diff
@@ -0,0 +1,151 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import com.github.mustachejava.Mustache;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Attachment;
+import org.hl7.fhir.dstu3.model.BaseReference;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.hl7.fhir.instance.model.api.IIdType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.NarrativeStatementTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.NarrativeStatementTemplateParameters.NarrativeStatementTemplateParametersBuilder;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DocumentReferenceToNarrativeStatementMapper {
+
+    private static final Mustache NARRATIVE_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_narrative_statement_template.mustache");
+    private static final String DEFAULT_ATTACHMENT_CONTENT_TYPE = "text/plain";
+    private final MessageContext messageContext;
+
+    public String mapDocumentReferenceToNarrativeStatement(final DocumentReference documentReference) {
+        final String narrativeStatementId = messageContext.getIdMapper()
+            .getOrNew(ResourceType.DocumentReference, documentReference.getId());
+
+        final NarrativeStatementTemplateParametersBuilder builder = NarrativeStatementTemplateParameters.builder()
+            .narrativeStatementId(narrativeStatementId)
+            .availabilityTime(getAvailabilityTime(documentReference));
+
+        final Optional<Attachment> contentAttachment = documentReference.getContent()
+            .stream()
+            .map(DocumentReference.DocumentReferenceContentComponent::getAttachment)
+            .findFirst();
+
+        if (contentAttachment.isPresent() && contentAttachment.get().hasTitle()) {
```

Comment:
I think that we will alway output the `<reference ...>` regardless if the attachment is "Normal" or "Absent".

the presence of content.attachment.title should just
* trigger creating the "AbsentAttachment" prefixed filename
* cause an extra "Absent Attachment: title..." text to appear in the pertinentInformation text
* ignore the attachment.contentType and always use text/plain for absent attachments

Also, if the DocumentReference is missing either its Content, Attachment, or contentType we should throw an exception - these are all required.

@dmcavenue1 have I read this spec correctly here?

Reply:
@mattd-kainos Yes - that looks correct to me

Reply:
Thanks - I will update this 😄

Reply:
Are we supporting multiple attachments @mattd-kainos @dmcavenue1?

Reply:
@n-r-b No 1 attachment per DocumentReference and NarrativeStatement. Thanks

Reply:
Great, thanks @dmcavenue1

Reply:
The attachment logic has now been updated, hopefully I've not missed anything